### PR TITLE
github: Move to a composite action for llvm-project builds

### PR DESCRIPTION
**Pull request by nathanchance**
This allows us to share the steps to build the container between the
different epoch workflows, which helps keeps things in sync as much as
possible.

As part of this change, pull the hardcoded tag out of test-clang.sh, so
that in the future, other stages can be tested.

### CODE REVIEW

**Review by nickdesaulniers — APPROVED** [review]
LGTM, but @compnerd understands composite actions more than I do; consider waiting for his review before merging.

Happy to see that idea of using composite actions being useful in additional places, and I'm glad I asked you guys for your suggestions on improvements, since I didn't know about composite actions.

Thanks for the patch.

File: .github/actions/build-test-llvm-project/action.yml
Comment (compnerd) [inline .github/actions/build-test-llvm-project/action.yml:22]: This is an odd way to build, but I see that its just a shuffle, so :man_shrugging:

**Review by compnerd — APPROVED** [review]

**Comment by nathanchance** [comment]
I'll merge this now and address any issues in a follow up.

**Pull request merged**
